storage/storage.cpp

```diff
@@ -1269,6 +1278,22 @@ bool Storage::HasLatestVersion(CountryId const & countryId) const
   return CountryStatusEx(countryId) == Status::OnDisk;
 }
 
+bool Storage::IsAllowedToEditVersion(CountryId const & countryId) const
+{
+  auto const status = CountryStatusEx(countryId);
+  switch (status) 
+  {
+    case Status::OnDisk: return true;
+    case Status::OnDiskOutOfDate:
+    {
+      auto const localFile = GetLatestLocalFile(countryId);
+      ASSERT(localFile, ("Local file shouldn't be nullptr."));
+      return m_currentVersion - localFile->GetVersion() < kOutdatedVersionDiffLimit;
```

Comment:
How does it work? For example `250101 - 241212 = 8889` but the difference is less than a month between these dates.

Reply:
Thanks! Replace this check with the timeInterval.
Also, I added a check to compare the current map version with the current time (now) to avoid situations where devices without a disconnected from the internet for a long time can update the map (6mo).